lib/oedipus_lex.rb

```diff
@@ -44,14 +44,7 @@ def to_ruby state, predicates, exclusive
         start_state == state or
         (state.nil? and predicates.include? start_state)
 
-      cond =
-        if exclusive or not start_state then
-          "when text = ss.scan(#{regexp}) then"
-        elsif start_state =~ /^:/ then
-          "when (state == #{start_state}) && (text = ss.scan(#{regexp})) then"
-        else
-          "when #{start_state} && (text = ss.scan(#{regexp})) then"
-        end
```

Comment:
It *looks* like I have tests for all 3 states... but your code below breaks it into 6 different states w/o additional tests? Is this accurate? If so, is this ok?